### PR TITLE
Fetch missing terms at set selection

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**

### CODE REVIEW

File: src/contributors/MetricsTableContributor.ts [hunk 1]
```diff
@@ -313,7 +328,63 @@ export class MetricsTableContributor extends Contributor {
...
+            const missingRows = [];
+            const dataRows = new Set(...this.computableResponse?.metricsResponse.map(r => r.keys));
+            termsSet.forEach(term => {
+                if (!dataRows.has(term)) {
+                    missingRows.push(term);
+                }
+            });
+            if (missingRows.length > 0) {
+                this.isDataUpdating = true;
+                return forkJoin(this.table.vectors.map(v =>
```
Comment (MAudelGisaia) [inline src/contributors/MetricsTableContributor.ts:345]: why do you need to return something as you subscribe into the method ? Its to unsubscribe  ?

File: src/models/mock-metrics.ts [hunk 2]
```diff
@@ -285,14 +285,14 @@ const aggregationResponseList: MetricsTableResponse[] =  [{
...
+//     columns: [
+//         {collection: 'toto', metric: ArlasApiMetric.CollectFctEnum.AVG, field: 'metadata_ObservationContext_eo_opt_cloudCoverPercentage'},
+//         {collection: 'toto', metric: ArlasApiMetric.CollectFctEnum.SUM, field: 'metadata_ObservationContext_eo_opt_cloudCoverPercentage'},
+//         {collection: 'titi', metric: ArlasApiMetric.CollectFctEnum.AVG, field: 'metadata_ObservationContext_eo_opt_cloudCoverPercentage'},
+//         {collection: 'titi', metric: ArlasApiMetric.CollectFctEnum.AVG, field: 'metadata_tristesse'},
+//         {collection: 'titi', metric: ArlasApiMetric.CollectFctEnum.SUM, field: 'metadata_heureux'},
+//         {collection: 'titi', metric: 'count', field: ''} // quizz du field ??? comment on sait que c'est le bon count
+//     ],
+//     metricsResponse: aggregationResponseList
+// };
```
Comment (MAudelGisaia) [inline src/models/mock-metrics.ts:298]: you can remove thus part. It was only for test

**Review by MAudelGisaia — APPROVED** [review]

File: src/contributors/MetricsTableContributor.ts (hunk 1 above)
Comment (MohamedHamouGisaia) [reply src/contributors/MetricsTableContributor.ts]: you are right, fixing

File: src/models/mock-metrics.ts (hunk 2 above)
Comment (MohamedHamouGisaia) [reply src/models/mock-metrics.ts:298]: I'll remove it in the final pr

**Pull request merged**